### ISSUE TITLE
grammar issue

### ISSUE THREAD

**Issue by avarices** [TypeStrong/atom-typescript#449]
I noticed that while in a class if you use assign a variable without a semicolon and declare a method with a binding, say `my_method(): string { }` the method name and type "string" will not be highlighted

Example for you guys to plug in in case the issue was just for me. 

```
class Test {
    constructor() {
    }
    breaker() {
       let a = "I broke it"
    }
    inflicted(): string {
       // "string" doesn't have proper highlighting
    }
}
```

What it looks like for me before:
![before]
(http://i.imgur.com/nexhqCA.png)

And after:
![after]
(http://i.imgur.com/8NQh9Ez.png)

I fixed it by adding a "?" next to the semicolon in the typerscript.cson file on line 33. It seemed to fix it.
Line 33 before: `'end': '(;)|\\b(in)\\b|(?<!,|^)$`
Line 33 now: `'end': '(;?)|\\b(in)\\b|(?<!,|^)$'`

**I haven't done extensive testing to see if it covers everything, but it fixed the files that were not properly highlighted for me**

I would also like to add that it it seemed properly highlighted in v4.8.3, but I'm not entirely sure.

**Comment by Cu3PO42** [comment]
Thanks for reporting this issue! Funnily enough I was just in the process of fixing this. Unfortunately your fix can't be used, it introduces other problems.

![screen shot 2015-07-10 at 00 08 24](https://cloud.githubusercontent.com/assets/5256300/8608075/ea2fde58-2697-11e5-823f-84678bb2cd43.png)

As you can see it breaks highlighting of types, etc. Essentially what is happening is that you create a RegEx that matches an empty string, thereby terminating the rule at the earliest possibility, not highlighting types, etc. I will properly fix this and release a new version with a couple of minor tweaks later.

Just out of curiosity, what theme is that? It looks awesome.

**Comment by avarices** [comment]
Ah, I see. I had a feeling that my proposal may have introduce new errors due to not properly testing it. I'm glad to hear a proper fix is coming. As far as the theme goes, thank you. I made it when I used Sublime Text 3. I based it off of Espresso Libre. Here is a download link if you want to try it out. https://www.dropbox.com/s/d6hs9p588dtd16j/extremis.tar.gz?dl=0

*I used apm to convert it (new to Atom), so it may not be as easily modified as say, atom dark*

**Comment by Cu3PO42** [comment]
I have a proper fix in #450, this will publish soon. The relevant change is actually in line 102 ;)

Thanks for a link to the theme! I yet have to port my theme from vim...

**Comment by avarices** [comment]
I added the fix, thanks for the quick reply and fix as well. Glad to hear the fix is it's going to be publishing soon! You're welcome for the theme, it's very green oriented, haha.

**Comment by Cu3PO42** [comment]
Fix is in 5.0.18!

**Comment by avarices** [comment]
Nice! I updated to 5.0.18 and all seems good. I came across another grammar issue, existing before the v5.0.18 update. I was wondering if a fix  for `for of` loops is coming soon as well? ES6 added `for of` loops for arrays, but it seems when you use `of` instead of `in`, the highlighting for the scope is removed, not sure if this should be a separate issue, I'm new to all of this, it's my first GitHub account. 

Self testing example 

```javascript
class Test {
    constructor() {
    }
    breaker() {
      let example_array = [[0, 1], [0, 2]]
      for (let [index, position] of example_array) {
          // all grammar in here seems to be gray
          let position_plus_one = position + 1
          console.log(position_plus_one)
      }
    }
}
```

Here is what it looks with `of`
![for_of_loss_of_highlighting]
(http://i.imgur.com/dxFOxcB.png)

And with `in` instead of `of`
![working_with_in]
(http://i.imgur.com/pJQUaFq.png)

The second example was just me testing if the issue was just the `of` keyword, seems to look that way. Extensive testing not done! Haha.
